storage/innobase/handler/ha_innodb.cc

```diff
@@ -16319,15 +16319,17 @@ innodb_buffer_pool_evict_uncompressed(void)
 			ut_ad(block->page.in_LRU_list);
 
 			mutex_enter(&block->mutex);
-			if (!buf_LRU_free_page(&block->page, false)) {
-				mutex_exit(&block->mutex);
-				all_evicted = false;
-			} else {
-				mutex_exit(&block->mutex);
+			all_evicted = buf_LRU_free_page(&block->page, false);
```

Comment:
No test case because the problem was originally reveled by existing `xtradb_compressed_columns_consistency_barracuda_compressed`, right?